strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/artifact/locator/ArtifactDirectoryLocator.java

```diff
@@ -22,51 +20,43 @@
     private ArtifactDirectoryOperation operation;
 
     /**
-     * The basedir to start analyzing from. Define this only,
-     * if you need to pass null values for operation.storage and operation.repository.
+     * The basedir to start analyzing from. Define this only, if you need to pass null values for operation.storage and
+     * operation.repository.
      */
-    private String basedir;
-
+    private RepositoryPath basedir;
 
     public void locateArtifactDirectories()
-            throws IOException
+        throws IOException
     {
         long startTime = System.currentTimeMillis();
 
-        Files.walk(Paths.get(getStartingPath()))
+        Files.walk(getStartingPath())
              .filter(Files::isDirectory)
              // Skip directories which start with a dot (like, for example: .index)
-             .filter(path -> !path.getFileName().startsWith("."))
+             .filter(path -> !(path.getNameCount() == 0 ? path.toString() : path.getName(path.getNameCount() - 1)).toString().startsWith("."))
              // Note: Sorting can be expensive:
              .sorted()
-             .forEach(operation::execute);
+             .forEach(this::execute);
 
         long endTime = System.currentTimeMillis();
 
         logger.debug("Executed (cache: " + operation.getVisitedRootPaths().size() + ")" +
-                     " visits in " + (endTime - startTime) + " ms.");
+                " visits in " + (endTime - startTime) + " ms.");
 
         getOperation().getVisitedRootPaths().clear();
     }
 
-    public String getStartingPath()
+    public RepositoryPath getStartingPath()
```

Comment:
@carlspring can you check that the original idea of this method is not distorted.